narayana-spring-boot-core/src/main/java/me/snowdrop/boot/narayana/core/jdbc/AbstractXADataSourceWrapper.java

```diff
@@ -25,32 +25,37 @@
 import org.springframework.boot.jdbc.XADataSourceWrapper;
 
 /**
- * {@link XADataSourceWrapper} implementation that uses {@link NarayanaDataSource} to wrap an {@link XADataSource}.
+ * An abstract {@link XADataSourceWrapper} implementation which creates a {@link XAResourceRecoveryHelper} capable to
+ * recovery transactions of the provided {@link XADataSource} and then delegates the actual data source wrapper to its
```

Comment:
This doesn't parse very well… 

Reply:
I would say it should be `capable of recovering transactions of the provided`

Reply:
Duh :)